editor/src/components/canvas/controls/select-mode/select-mode.spec.browser.ts

```diff
@@ -0,0 +1,338 @@
+      fireEvent(
+        canvasControlsLayer,
+        new MouseEvent('mousedown', {
+          detail: 1,
+          bubbles: true,
+          cancelable: true,
+          metaKey: false,
+          clientX: areaControlBounds.left + 20,
+          clientY: areaControlBounds.top + 20,
+          buttons: 1,
+        }),
+      )
+      fireEvent(
+        canvasControlsLayer,
+        new MouseEvent('mousedown', {
+          detail: 2,
+          bubbles: true,
+          cancelable: true,
+          metaKey: false,
+          clientX: areaControlBounds.left + 20,
+          clientY: areaControlBounds.top + 20,
+          buttons: 1,
+        }),
+      )
+      await domFinished
+      await dispatchDone
+    })
+    await waitForAnimationFrame()
+
+    const selectedViews4 = renderResult.getEditorState().editor.selectedViews
+
+    await act(async () => {
+      const domFinished = renderResult.getDomReportDispatched()
+      const dispatchDone = renderResult.getDispatchFollowUpactionsFinished()
+      fireEvent(
+        canvasControlsLayer,
+        new MouseEvent('mousedown', {
+          detail: 1,
+          bubbles: true,
+          cancelable: true,
+          metaKey: false,
+          clientX: areaControlBounds.left + 20,
+          clientY: areaControlBounds.top + 20,
+          buttons: 1,
+        }),
+      )
+      fireEvent(
+        canvasControlsLayer,
+        new MouseEvent('mousedown', {
+          detail: 2,
+          bubbles: true,
+          cancelable: true,
+          metaKey: false,
+          clientX: areaControlBounds.left + 20,
+          clientY: areaControlBounds.top + 20,
+          buttons: 1,
+        }),
+      )
+      await domFinished
+      await dispatchDone
+    })
+    await waitForAnimationFrame()
+
+    const selectedViews5 = renderResult.getEditorState().editor.selectedViews
+
+    await act(async () => {
+      const domFinished = renderResult.getDomReportDispatched()
+      const dispatchDone = renderResult.getDispatchFollowUpactionsFinished()
+      fireEvent(
+        canvasControlsLayer,
+        new MouseEvent('mousedown', {
+          detail: 1,
+          bubbles: true,
+          cancelable: true,
+          metaKey: false,
+          clientX: areaControlBounds.left + 20,
+          clientY: areaControlBounds.top + 20,
+          buttons: 1,
+        }),
+      )
+      fireEvent(
+        canvasControlsLayer,
+        new MouseEvent('mousedown', {
+          detail: 2,
+          bubbles: true,
+          cancelable: true,
+          metaKey: false,
+          clientX: areaControlBounds.left + 20,
+          clientY: areaControlBounds.top + 20,
+          buttons: 1,
+        }),
+      )
+      await domFinished
+      await dispatchDone
+    })
+    await waitForAnimationFrame()
+
+    const selectedViews6 = renderResult.getEditorState().editor.selectedViews
+
+    expect(selectedViews1).toEqual([TP.instancePath(['utopia-storyboard-uid', 'scene-aaa'], ['a'])])
```

Comment:
I think it's worth moving these tests after their respective selection so that the test can fail early if there's an issue, and so that if the test locks up it will be easier to trace where it locked up.

Reply:
fixed